drivers/console/ipm_console.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <init.h>
+#include <drivers/ipm.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ipm_console, CONFIG_IPM_LOG_LEVEL);
+
+static struct device *ipm_console_device;
+
+static int console_out(int c)
+{
+	/* Send character with busy wait */
+	ipm_send(ipm_console_device, 1, c & ~BIT(31), NULL, 0);
```

Comment:
OK, excellent example of why you want queuing in this driver.  This is going to stall waiting on the other side to take an interrupt and clear the busy flat for... every... byte... of... log... output...  Some linux driver developer is going to be coming at you with a shiv made from the shards of a smartphone screen.

FWIW: accumulating a local line buffer and flushing when it gets full or on newline would be a very reasonable workaround too.

Reply:
I took it initially from `drivers/console/ipm_console_sender.c` https://github.com/zephyrproject-rtos/zephyr/blob/49bf8b1d544fa932f1d012d2c9252fdf1021b167/drivers/console/ipm_console_sender.c#L28

I think we can just send without busy wait. If we get BUSY we can just drop the character.

For the testing it was easier to set this flag to verify that it works.

Reply:
I have now immediate log with line buffer and another option using message queue msgq.